persistence/persistence-spark/src/main/java/org/cbioportal/persistence/spark/ClinicalDataSparkRepository.java

```diff
@@ -0,0 +1,280 @@
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import scala.collection.JavaConverters;
+import scala.collection.Seq;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static org.apache.spark.sql.functions.*;
+
+
+@Component
+@Qualifier("clinicalDataSparkRepository")
+public class ClinicalDataSparkRepository implements ClinicalDataRepository {
+
+    @Autowired
+    private SparkSession spark;
+
+    @Value("${data.parquet.folder}")
+    private String PARQUET_DIR;
+
+    private static Log logger = LogFactory.getLog(ClinicalDataSparkRepository.class);
+    private static final String MUTATION_COUNT = "MUTATION_COUNT";
+    private static final String FRACTION_GENOME_ALTERED = "FRACTION_GENOME_ALTERED";
+
+    @Override
+    public List<ClinicalData> getAllClinicalDataOfSampleInStudy(String studyId, String sampleId, String attributeId, String projection, Integer pageSize, Integer pageNumber, String sortBy, String direction) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public BaseMeta getMetaSampleClinicalData(String studyId, String sampleId, String attributeId) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<ClinicalData> getAllClinicalDataOfPatientInStudy(String studyId, String patientId, String attributeId, String projection, Integer pageSize, Integer pageNumber, String sortBy, String direction) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public BaseMeta getMetaPatientClinicalData(String studyId, String patientId, String attributeId) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<ClinicalData> getAllClinicalDataInStudy(String studyId, String attributeId, String clinicalDataType, String projection, Integer pageSize, Integer pageNumber, String sortBy, String direction) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public BaseMeta getMetaAllClinicalData(String studyId, String attributeId, String clinicalDataType) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<ClinicalData> fetchAllClinicalDataInStudy(String studyId, List<String> ids, List<String> attributeIds, String clinicalDataType, String projection) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public BaseMeta fetchMetaClinicalDataInStudy(String studyId, List<String> ids, List<String> attributeIds, String clinicalDataType) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public List<ClinicalData> fetchClinicalData(List<String> studyIds, List<String> sampleIds, List<String> attributeIds,
+                                                String clinicalDataType, String projection) {
+        List<Dataset<Row>> res = new ArrayList<>();
+        for (String studyId : new HashSet<>(studyIds)) {
```

Comment:
Instead of iterating studies to read parquet file why not read all files at a time. 
something like `spark.read().parquet("file1", "file2",...)`. or may be a path pattern matching. not sure if it's supported in java spark library

Reply:
It is possible to load all files at a time, like I do in fetchClinicalDataCounts() method. I iterated through to set ClinicalData.studyId.

Reply:
that should work

Reply:
which one would work? Do you prefer I change it to load all files at once or iterate through

Reply:
load all files at once

Reply:
As I refactor, I see that setting the studyId is necessary for StudyViewController.java line 387. clinicalDataMap = clinicalDataList.stream().collect(Collectors.groupingBy(ClinicalData::getSampleId, 
                Collectors.groupingBy(ClinicalData::getStudyId)));

Reply:
you can try adding new column `STUDY_ID` to dataset

Reply:
sorry, I don't follow you. if i'm merging all studies, then how would i know which studyId to set in ClinicalData?

Reply:
you could get study id from the input file name. something like this https://stackoverflow.com/a/40849244.

`.withColumn("input_file_name", getStudyId(input_file_name()));`

you will have to implement `getStudyId` to extract study if from path

Reply:
sure. updated the code.

Reply:
nice. just curious, was there any performance improvement with this approach?

Reply:
I added https://user-images.githubusercontent.com/2739448/61876051-f12dbc80-aeb9-11e9-90e9-5432ae913a69.png in the other comment. Doesn't seem to have much impact on performance for 2 studies.